File: 5-helo/helo.r
#!/usr/bin/env -S -- Rscript

argv <- commandArgs()
location <- argv[4]
if (!startsWith(location, "--file=")) {
  stop()
}

file <- sub("^--file=", "", location)
script <- normalizePath(file)

setwd(dirname(file))
cat(paste0("HELO :: VIA -- ", basename(file), "\n"))

if (system2(c("bat", "--", script)) != 0) {
  stop()
}

File: 5-helo/helo.py
#!/usr/bin/env -S -- PYTHONSAFEPATH= python3

from os import chdir
from pathlib import Path
from subprocess import run

file = Path(__file__).resolve()
chdir(file.parent)

print(f"HELO :: VIA -- {file.relative_to(Path.cwd())}")
run(("bat", "--", file)).check_returncode()
